### PR TITLE
Add GitHub release workflow

### PR DESCRIPTION
**Pull request by mkuznyetsov**
### What does this PR do?
Add workflow for releasing che-theia on GitHub actions. Adapt make-release.sh script for this workflow

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    inputs:
+      version:
+        description: 'The version that is going to be released. Should be in format 7.y.z'
+        required: true
+        default: ''
+      forceRecreateTags:
+        description: If true, tags will be recreated. Use with caution
+        required: false
+        default: 'false'
+      pushToNpmJs:
```
Comment (azatsarynnyy) [inline .github/workflows/release.yml:23]: where it's used?

Comment (mkuznyetsov) [reply .github/workflows/release.yml:23]: oops, missed some stashed changes, now it's used down the workflow

Comment (mkuznyetsov) [reply .github/workflows/release.yml:23]: it'll probably never be used, but it's just gonna be there as an option to build image without pushing stuff to npmjs

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        description: If true, will push images to npmjs
```
Comment (benoitf) [inline .github/workflows/release.yml:24]:
```suggestion
        description: If true, will push packages to npmjs
```

**Review by benoitf — APPROVED** [review]

**Pull request merged**